code/game/objects/structures/stool_bed_chair_nest/bed.dm

```diff
@@ -103,7 +103,15 @@
 				"You unbuckle yourself from \the [src].",
 				"You hear metal clanking.")
 		playsound(src, 'sound/misc/buckle_unclick.ogg', 50, 1)
-		return TRUE
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(H.disabilities & ANEMIA)
+				if(world.time - H.lastAnemia <= 30 SECONDS)
```

Comment:
move this check up into the if statement on line 108

Reply:
done!